### ISSUE TITLE
More Question than Issue - Automatic File Size?

### ISSUE THREAD

**Issue by whitedavidp** [meefik/linuxdeploy#898]
It is not exactly clear to me what the automatic file size does/means. I can see that by setting this for a new install, a file size is somehow chosen/used. But I have several questions:

1) How is the file size determined/selected?

2) Does this mean that over time the file size may increase? If so how/why and to what limit?

3) On different partition formats, there will be different file size limits. When specifying a file size are the values in MB, GB, or what? If I specify a file size too large, what happens?

4) If the answer to (2) is yes, the file size may increase over time and if the file size expansion would exceed the maximum file size for a given partition type, what happens?

5) Once an install file is created, can the file size be manually changed if desired? If so, how to do this? For example, the automatic size selected a 2GB file. On my FAT32 USB disk, I have more free space and would like to take the file size up to maybe 4GB. But I don't want to do this if any automatic size change mechanism might make a change that ends up exceeding the max for FAT32 and this corrupting my install file.

Thanks for all!

**Comment by WilsonBradley** [comment]
Was wondering about this option also.. If loop file is on fat32 poartion there will always be a 4gb limit and myself am trying to get around this.

**Comment by Danct12** [comment]
Max. file size 2,147,483,647 bytes (2 GiB – 1) (without LFS
<https://en.wikipedia.org/wiki/Large_file_support>)
4,294,967,295 bytes (4 GiB – 1)[1]
<https://en.wikipedia.org/wiki/File_Allocation_Table#cite_note-GB4-1> (with
LFS <https://en.wikipedia.org/wiki/Large_file_support>)
274,877,906,943 bytes (256 GiB – 1) (only with FAT32+[35]
<https://en.wikipedia.org/wiki/File_Allocation_Table#cite_note-DRDOS_FAT+_R2-41>
)Basically you can't get around the 4 GB file size limit on a FAT32,
however to allow creation of image larger >= 4 GB, the SD Card partition
(where the loop file is stored) needs to use a file system that supports
it. You can use ext4.

Vào Th 7, 30 thg 3, 2019 vào lúc 03:11 Wilson K Bradley <
notifications@github.com> đã viết:

**Comment by whitedavidp** [comment]
Thanks fr confirming this. Do you know of any way to extend an existing loop file if currently on a FAT32 SD card and moved to ext4 or other that permits larger file size? Of must the loop file be re-created in the app?

**Comment by Danct12** [comment]
Yes, the loop file size can be increased without re-creating! Check this comment out:
https://github.com/meefik/linuxdeploy/issues/395#issuecomment-370231632

**Comment by whitedavidp** [comment]
That's great! Thanks for the pointer and help.

**Comment by Danct12** [comment]
Just a quick heads up, some Android version will not read the partitions formatted under ext4, however you can manually mount it in the terminal.
For list of partitions (both internal and external) for your device, you can `ls /dev/block`

**Comment by WilsonBradley** [comment]
I was not getting notifications of updates here..
Sorry for the late reply.

Yes, since then I have formatted my SD card exFat and am now able to have files bigger than 4gb, thus
For LinuxDeploy was able to make the loop file 8gb.  

Thanks
